notebook2/notebook/notebook.py

```diff
@@ -106,6 +107,24 @@ def get_name(line):
         "containing the name of the docker image to use for worker pods.") from e
 
 
+def requires_auth(for_page = True):
+    def auth(f):
+        @wraps(f)
+        def decorated(*args, **kwargs):
+            if 'user' not in session:
+                # Redirect to Login page here
+                if for_page is True:
+                    session['referrer'] = request.url
+                    return redirect(flask.url_for('login_page'))
+
+                return '', 401
```

Comment:
Could be 404 if we prefer giving less info. I slightly prefer that

Reply:
I don't think we should 404 on `/` to unauthenticated users. It'd be reasonable to 404 on a *resource* that the user doesn't own, but that would be authorization, we're doing authentication here. 401 seems like the right response.

Reply:
`/` is a carry from notebook1 (refers to notebook). So it's a GET to notebook resources.

Reply:
And I agree (I think); / should not return a 404, unless the entire site is private.